cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ConfiguratorTask.java

```diff
@@ -0,0 +1,124 @@
+import io.cdap.cdap.common.internal.worker.RunnableTaskContext;
+import io.cdap.cdap.common.io.Locations;
+import io.cdap.cdap.internal.app.ApplicationSpecificationAdapter;
+import io.cdap.cdap.internal.app.deploy.InMemoryConfigurator;
+import io.cdap.cdap.internal.app.deploy.pipeline.ConfiguratorConfig;
+import io.cdap.cdap.internal.app.runtime.artifact.ArtifactRepository;
+import io.cdap.cdap.internal.app.runtime.artifact.PluginFinder;
+import io.cdap.cdap.security.impersonation.EntityImpersonator;
+import io.cdap.cdap.security.impersonation.Impersonator;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * ConfiguratorTask is a RunnableTask for performing the configurator config.
+ */
+public class ConfiguratorTask implements RunnableTask {
+  private static final Gson GSON = ApplicationSpecificationAdapter.addTypeAdapters(new GsonBuilder()).create();
+  private static final Logger LOG = LoggerFactory.getLogger(ConfiguratorTask.class);
+
+
+  public ConfiguratorTask() {
+
+  }
+
+  @Override
+  public void run(RunnableTaskContext context) throws Exception {
+    Injector injector = Guice.createInjector(new ConfiguratorTaskModule());
+    Configurator configurator = injector.getInstance(Configurator.class);
+    configurator.run(context);
+  }
+
+  /**
+   * Configurator performs the configurator config.
+   */
+  public static class Configurator {
+    private final Impersonator impersonator;
+    private final PluginFinder pluginFinder;
+    private final ArtifactRepository artifactRepository;
+    private final LocationFactory locationFactory;
+
+    @Inject
+    public Configurator(Impersonator impersonator, PluginFinder pluginFinder,
+                        ArtifactRepository artifactRepository, LocationFactory locationFactory) {
+      this.impersonator = impersonator;
+      this.pluginFinder = pluginFinder;
+      this.artifactRepository = artifactRepository;
+      this.locationFactory = locationFactory;
+    }
+
+    public void run(RunnableTaskContext context) throws Exception {
+      ConfiguratorConfig config = GSON.fromJson(context.getParam(), ConfiguratorConfig.class);
+
+      // Getting the pipeline app from appfabric
+      LOG.info(String.format("Fetching artifact '%s' from app-fabric to create artifact class loader.",
+                             config.getArtifactId().getName()));
+      Location artifactLocation = Locations.getLocationFromAbsolutePath(
+        locationFactory, config.getArtifactLocationURI().getPath());
+      OutputStream outputStream = artifactLocation.getOutputStream();
+      InputStream artifactBytes = artifactRepository.getArtifactBytes(config.getArtifactId());
+      ByteStreams.copy(artifactBytes, outputStream);
+      outputStream.close();
```

Comment:
Never call close like this. Use the automatic resource release construct.

Reply:
done